### PR TITLE
[Bugfix] Add screening questions to CSV

### LINKED ISSUE #11028: 🐛 Missing screening questions in candidate CSV
## 🐛 Bug

The candidate CSV document does not include screening questions but should.

## 🦋 Expected Behaviour

Screening questions and answers should appear in the candidate CSV.

## 📋 Steps to Reproduce

1. Go to the pool candidates table
2. Download a candidate that has answered some screening questions
3. Observe they do not appear in the CSV

## 🙋‍♀️ Proposed Solution

- Prefix the general questions column header with "General question:"
- Add new columns for screening questions directly after general questions
- Add prefix "Screening question:" to differentiate from general questions

## ✅ Acceptance Criteria

- [ ] Screening questions added to candidate CSV directly following general questions
- [ ] General and Screening questions are prefixed in the CSV column headers

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #11028

## 👋 Introduction

Add screening questions and responses to the candidates CSV download. 

## 🧪 Testing

1. Have a candidate attached to a process with screening questions
2. Run `make queue-work`
3. Download CSV for candidate and confirm existence of added columns

### CODE REVIEW

File: api/app/Generators/CandidateProfileCsv.php [hunk 1]
```diff
@@ -218,15 +233,26 @@ public function generate()
...
             })->chunk(100, function ($pools) {
                 foreach ($pools as $pool) {
                     if ($pool->generalQuestions->count() > 0) {
                         foreach ($pool->generalQuestions as $question) {
-                            $this->questionIds[] = $question->id;
-                            $this->generatedHeaders['general_questions'][] = $question->question[$this->lang];
+                            $this->generalQuestionIds[] = $question->id;
+                            $this->generatedHeaders['general_questions'][] = $this->lang == 'en' ?
+                            'General question: '.$question->question[$this->lang] :
+                            'Frenchy string: '.$question->question[$this->lang];
```
Comment (vd1992) [inline api/app/Generators/CandidateProfileCsv.php:246]: What should be the string here?

General questions plural

image

image

File: api/app/Generators/CandidateProfileCsv.php [hunk 2]
```diff
@@ -218,15 +233,26 @@ public function generate()
...
+                        }
+                    }
+
+                    if ($pool->screeningQuestions->count() > 0) {
+                        foreach ($pool->screeningQuestions as $question) {
+                            $this->screeningQuestionIds[] = $question->id;
+                            $this->generatedHeaders['screening_questions'][] = $this->lang == 'en' ?
+                            'Screening question: '.$question->question[$this->lang] :
+                            'Frenchy string: '.$question->question[$this->lang];
```
Comment (vd1992) [inline api/app/Generators/CandidateProfileCsv.php:255]: What should be the string here?

Screening questions plural

image

image

Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php:255]: We've a similar string below that reads differently as a heads up

image

image

File: api/app/Generators/CandidateProfileCsv.php (hunk 1 above)
Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php:246]: @tristan-orourke 
Thoughts on both comments? (Not sure who to ask in specific)

Comment (tristan-orourke) [reply api/app/Generators/CandidateProfileCsv.php:246]: Stick with singular: "General question: {the actual question}"

File: api/app/Generators/CandidateProfileCsv.php (hunk 2 above)
Comment (tristan-orourke) [reply api/app/Generators/CandidateProfileCsv.php:255]: Stick with singular, and the index is not necessary in the csv header.

File: api/app/Generators/CandidateProfileCsv.php (hunk 1 above)
Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php:246]: Sorry for the confusion, but yeah, I meant French 😛

Comment (tristan-orourke) [reply api/app/Generators/CandidateProfileCsv.php:246]: I see. The singular of this is just `Question générale`

File: api/app/Generators/CandidateProfileCsv.php (hunk 2 above)
Comment (tristan-orourke) [reply api/app/Generators/CandidateProfileCsv.php:255]: Let's go with `Question de sélection#. And while we're at it, fix that other to match: `Question de sélection {index}`.

File: api/app/Generators/CandidateProfileCsv.php (hunk 1 above)
Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php]: Okay, added in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11093/commits/9b7967330169756d9ece240ec03f343fb488f8bb

File: api/app/Generators/CandidateProfileCsv.php (hunk 2 above)
Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php]: Thanks, I added/changed both in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11093/commits/9b7967330169756d9ece240ec03f343fb488f8bb

File: api/app/Generators/CandidateProfileCsv.php [hunk 3]
```diff
@@ -218,15 +233,26 @@ public function generate()
...
+                        }
+                    }
+
+                    if ($pool->screeningQuestions->count() > 0) {
+                        foreach ($pool->screeningQuestions as $question) {
+                            $this->screeningQuestionIds[] = $question->id;
+                            $this->generatedHeaders['screening_questions'][] = $this->lang == 'en' ?
+                            'Screening question: '.$question->question[$this->lang] :
+                            'Question de sélection: '.$question->question[$this->lang];
                         }
```
Comment (esizer) [inline api/app/Generators/CandidateProfileCsv.php:256]: https://github.com/GCTC-NTGC/gc-digital-talent/blob/e0381a2a9f6b7dff87e85b5151ed7167ef8e8d27/api/lang/fr/headings.php#L60

bekphnqftcb41

Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php:256]: "Singular verse plural", remarked the 🟢 👺

**Review by esizer — CHANGES_REQUESTED** [review]
Looking good, just need to update how we access the languages to use the laravel localization.

File: api/app/Generators/CandidateProfileCsv.php [hunk 4]
```diff
@@ -218,15 +233,26 @@ public function generate()
...
             })->chunk(100, function ($pools) {
                 foreach ($pools as $pool) {
                     if ($pool->generalQuestions->count() > 0) {
                         foreach ($pool->generalQuestions as $question) {
-                            $this->questionIds[] = $question->id;
-                            $this->generatedHeaders['general_questions'][] = $question->question[$this->lang];
+                            $this->generalQuestionIds[] = $question->id;
+                            $this->generatedHeaders['general_questions'][] = $this->lang == 'en' ?
+                            'General question: '.$question->question[$this->lang] :
+                            'Question générale: '.$question->question[$this->lang];
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Generators/CandidateProfileCsv.php]: Ah okay, thanks for the details. Yes, I shall leverage the other PR's additions. 
Okay, done in 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/11093/commits/238c75fa765a7b898f1eb2a7b260177c6cbf8a11
https://github.com/GCTC-NTGC/gc-digital-talent/pull/11093/commits/b8ca4a4badb37821f6be76f9b09a961ab77bdc38

So like that?

**Review by esizer — APPROVED** [review]
Nice! :sunglasses:

**Pull request merged**